avro-fastserde/src/main/java/com/linkedin/avro/fastserde/PrimitiveFloatList.java

```diff
@@ -60,44 +75,73 @@ public PrimitiveFloatList(Collection<Float> c) {
    * @throws IOException on io errors
    */
   public static Object readPrimitiveFloatArray(Object old, Decoder in) throws IOException {
-    long l = in.readArrayStart();
-    if (l > 0) {
-      PrimitiveFloatList array = (PrimitiveFloatList) newPrimitiveFloatArray(old, (int) l);
+    long length = in.readArrayStart();
+    long totalLength = 0;
+
+    if (length > 0) {
+      PrimitiveFloatList array = (PrimitiveFloatList) newPrimitiveFloatArray(old);
+      int cnt = 0;
+
       do {
-        for (long i = 0; i < l; i++) {
-          array.addPrimitive(in.readFloat());
+        ByteBuffer byteBuffer;
+        // Allocate ByeBuffer of size array length * Float.BYTES to hold the array values
+        if (array.byteBuffers.size() > cnt && (array.byteBuffers.get(cnt).limit() > (length * FLOAT_SIZE))) {
+          byteBuffer = array.byteBuffers.get(cnt);
+          byteBuffer.clear();
+        } else {
+          byteBuffer = ByteBuffer.allocate((int)length * FLOAT_SIZE).order(ByteOrder.LITTLE_ENDIAN);
         }
-        l = in.arrayNext();
-      } while (l > 0);
+        in.readFixed(byteBuffer.array(), 0, (int)(length * FLOAT_SIZE));
+        array.byteBuffers.add(cnt++, byteBuffer);
+        totalLength += length;
+        length = in.arrayNext();
+      } while (length > 0);
+
+      array.byteBufferCount = cnt;
+      setupElements(array, (int)totalLength);
       return array;
     } else {
-      return newPrimitiveFloatArray(old, 0);
+      return new PrimitiveFloatList(0);
+    }
+  }
+
+  private static void setupElements(PrimitiveFloatList list, int total_size) {
```

Comment:
Can we say avoid the underscore syntax in favor of camelCase, which is more typical in Java?

Reply:
Done